src/Composition/ForceElements/SpringDamperPtP.jl

```diff
@@ -1,28 +1,60 @@
 
 """
-force = SpringDamperPtP(; obj1, obj2, stiffness, damping)
+    force = SpringDamperPtP(; obj1, obj2,
+        nominalLength          = 0.0,
+        nominalForce           = 0.0,
+        stiffness              = 0.0,
+        stiffnessForceFunction = nothing,
+        damping                = 0.0,
+        dampingForceFunction   = nothing )
 
 Return a `force` acting as point-to-point parallel spring-damper between
-`obj1::`[`Object3D`](@ref) and `obj2::`[`Object3D`](@ref). Values
-`stiffness` and `damping` define the stiffness resp. damping coefficient.
+`obj1::`[`Object3D`](@ref) and `obj2::`[`Object3D`](@ref).
+
+# Arguments
+
+- `nominalLength`: Nominal length, i.e. distance between `obj1` and `obj2`
+  where deflection of stiffness force is zero.
+- `nominalForce`: Nominal force, i.e. force that acts when stiffness and
+  damping forces are zero. Positive values represent tensil forces.
+- `stiffness`: Linear stiffness coefficient of spring. Ignored if
+  `stiffnessForceFunction` is defined.
+- `stiffnessForceFunction`: Univariate function which computes the
+  stiffness force dependent of the stiffness force deflection.
+- `damping`: Linear damping coefficient of damper. Ignored if
+  `dampingForceFunction` is defined.
+- `dampingForceFunction`: Univariate function which computes the
+  damping force dependent of the damper velocity.
 """
 mutable struct SpringDamperPtP <: Modia3D.AbstractForceElement
 
     obj1::Modia3D.AbstractObject3D
     obj2::Modia3D.AbstractObject3D
 
+    nominalLength::Float64
+    nominalForce::Float64
+
     stiffness::Float64
+    stiffnessForceFunction::Union{Function, Nothing}
```

Comment:
For performance reasons avoid `Union` in struct fields, please. In this case you can help yourself with a flag. See e.g. object3D.jl:105.

Reply:
Okay, so what should the constructor assign to `stiffnessForceFunction` if no function is defined by the user?

Reply:
You can call me. I'll explain it to you.

Reply:
I would suggest to provide two force elements: One with linear spring and linear damper (no function), and one with given functions.

Reply:
Ok, this is an even better idea!

Reply:
I don't think so. A common use case is to have a function for the stiffness force and linear damping. However, the reversed case might occur, too. And having separate elements would mean to have duplicate code.

Reply:
The current design has also the drawback that one can provide parameters that do not have an effect (e.g., c,d and the functions can be given and c,d has then no effect). One has to also think about the graphical user interface, that should be as simple as possible. I therefore propose to have only a linear spring/damper element and an additional lineForce element which consists of one function that gets the relative distance and the relative velocity as input arguments and returns a line force.

Reply:
Please take a look at the last commit d5fa3ea: I have changed it to `springForceLaw::Union{Float64, Function}` so that one parameter can be used for stiffness values as well as spring force functions.